partner_communication/models/communication_job.py

```diff
@@ -468,42 +469,32 @@ def write(self, vals):
     #                             PUBLIC METHODS                             #
     ##########################################################################
     def send(self):
-        """Executes the job."""
+        """
+        Executes the job asynchronously for each communication task.
+        This ensures that each job is processed independently in the queue.
+        """
+        # Filter "pending" tasks
         todo = self.filtered(
             lambda j: j.state == "pending"
             and not (j.need_call == "before_sending" and j.activity_ids)
         )
+        todo.write({"state": "processing"})
 
-        # Filter jobs for SMS
-        sms_jobs = self.filtered(lambda j: j.send_mode == "sms")
-        sms_jobs.send_by_sms()
+        # Process SMS jobs asynchronously
+        sms_jobs = todo.filtered(lambda j: j.send_mode == "sms")
+        if sms_jobs:
+            sms_jobs.send_by_sms()
 
+        # Process physical print jobs asynchronously
         to_print = todo.filtered(lambda j: j.send_mode == "physical")
-        for job in todo.filtered(lambda j: j.send_mode in ("both", "digital")):
-            origin = self.env.context.get("origin")
-            # if we print first in a communication with send_mode == both
-            if origin == "both_print" and job.send_mode == "both":
-                job.send_mode = "digital"
-                return job._print_report()
-
-            state = job._send_mail()
-            if job.send_mode != "both":
-                job.write(
-                    {
-                        "state": state,
-                        "sent_date": state != "pending" and fields.Datetime.now(),
-                    }
-                )
-            else:
-                # Job was sent by e-mail and must now be printed
-                job.send_mode = "physical"
-                job.refresh_text()
-
-            # if the call must be done after the sending, an activity is scheduled
-            if job.need_call == "after_sending":
-                job.with_user(job.user_id.id).schedule_call()
         if to_print:
-            return to_print._print_report()
+            to_print._print_report()
+
+        # Process email jobs (digital or both) asynchronously
+        email_jobs = todo.filtered(lambda j: j.send_mode in ("both", "digital"))
+        for job in email_jobs:
+            job.with_delay()._process_email_job_asynchronous()
```

Comment:
Add the channel for grouping jobs and making sure no concurrent jobs are running in parallel.
```suggestion
            job.with_delay(channel="root.partner_communication")._process_email_job_asynchronous()
```